src/ray/core_worker/task_manager.cc

```diff
@@ -341,6 +342,7 @@ bool TaskManager::ResubmitTask(const TaskID &task_id, std::vector<ObjectID> *tas
         RAY_CHECK(it->second.num_retries_left == -1);
       }
       spec = it->second.spec;
+      task_entry = &(it->second);
```

Comment:
This is discourage in C++, especially under concurrent use case.
When a map changes (i.e. entry deleted or inserted), all key-value pairs are subjected to move to another memory address, which means the pointer `task_entry` gets invalidated.

The recommended way is to use unique pointer as value, so you could get a stable pointer of value.

Reply:
move the discussion to https://github.com/ray-project/ray/pull/52770

Reply:
happy to learn some new C++ knowledge!